### PR TITLE
Sign Up Flow API

### LINKED ISSUE #31: Design endpoints for SignUp
User's sign-up is multiple step process. To accomplish its on backend part we need to implement next endpoints: 
1. `POST /signup` - this endpoint handles user registration requests, sent from sign-up form.
2. `GET /email/available` - this endpoint compares user's entered email with existing emails and displays an error if there is the same in DB.
3. `GET /users/emailVerificationToken/verify` - this endpoint gives a unique token to user which is needed to complete the registration and verify email.

### PR DESCRIPTION
**Pull request by Krimyllex**
Closes #31

### CODE REVIEW

File: api/ims/ims-api.yaml [hunk 1]
```diff
@@ -40,79 +38,138 @@ paths:
...
+          description: User's email
         password:
           type: string
           format: password
-          description: Users password
+          description: User's password
+      required:
+        - email
+        - password
+    SignupResponse:
```
Comment (taaraora) [inline api/ims/ims-api.yaml:157]: We don't need such a message. status code 200 is enough to understand that the signup request has been processed successfully.

File: api/ims/ims-api.yaml [hunk 2]
```diff
@@ -40,79 +38,138 @@ paths:
...
+      type: object                                    
+      properties:
+        message: 
+          type: string
+          description: Sign up request successful, to complete registration confirm your email with token 
+      required:
+       - message
+    SignupResponseWithToken:
```
Comment (taaraora) [inline api/ims/ims-api.yaml:165]: There is no point in having such a response.

File: api/ims/ims-api.yaml [hunk 3]
```diff
@@ -40,79 +38,138 @@ paths:
...
+          schema:
+            type: string
+          description: Token that you need to complete the registration
+      responses:
+        '200':
+          description: Registration completed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SignupResponseWithToken'
```
Comment (taaraora) [inline api/ims/ims-api.yaml:110]: It is enough to have just status code 200.

File: api/ims/ims-api.yaml [hunk 4]
```diff
@@ -40,79 +38,138 @@ paths:
...
+          description: User's email
         password:
           type: string
           format: password
-          description: Users password
-    LoginRequest:
-      type: object
+          description: User's password
       required:
+        - firstName
```
Comment (taaraora) [inline api/ims/ims-api.yaml:139]: why did you add these fields in the required section of this entity?
`
- firstName
- lastName
`

File: pkg/signup/repository/mssql/SignUp_init.down.sql [hunk 5]
```diff
@@ -0,0 +1,4 @@
+--Used MS SQL Server Management Studio
```
Comment (taaraora) [inline pkg/signup/repository/mssql/SignUp_init.down.sql:1]: please remove SQL code from this PR

File: api/ims/ims-api.yaml (hunk 4 above)
Comment (Krimyllex) [reply api/ims/ims-api.yaml:139]: I did not add them into the required section. They have already been there. And, as I know, you added them.

Comment (Krimyllex) [reply api/ims/ims-api.yaml:139]: And this is not my block of code.
